File: Greedy/55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int c = nums[0];
        for(int i = 1; i < nums.length; i++){
            if(c < i){
                return false;
            }
            c = Math.max(c, nums[i]+i);
        }
        return true;
    }
}

File: 55-jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int cj = 0;
        int i = 0;
        for(int i = 0; i < nums.size(); i++){
            if(i > cj) return false;
            cj = max(cj, nums[i]+i);
        }
        return true;
    }
};